Firestore/core/test/unit/local/local_store_test.cc

```diff
@@ -1454,6 +1479,195 @@ TEST_P(LocalStoreTest, HandlesPatchMutationWithTransformThenRemoteEvent) {
+       HandlesMergeMutationWithTransformationThenBundledDocuments) {
+  core::Query query = Query("foo");
+  AllocateQuery(query);
+
+  WriteMutation(
+      testutil::MergeMutation("foo/bar", Map(), std::vector<model::FieldPath>(),
+                              {testutil::Increment("sum", Value(1))}));
+
+  FSTAssertContains(
+      Doc("foo/bar", 0, Map("sum", 1), DocumentState::kLocalMutations));
+  FSTAssertChanged(
+      Doc("foo/bar", 0, Map("sum", 1), DocumentState::kLocalMutations));
+
+  ApplyBundledDocuments({Doc("foo/bar", 1, Map("sum", 1337))});
+  FSTAssertChanged(
+      Doc("foo/bar", 1, Map("sum", 1), DocumentState::kLocalMutations));
+  FSTAssertContains(
+      Doc("foo/bar", 1, Map("sum", 1), DocumentState::kLocalMutations));
+
+  DocumentKeySet expected_keys({Key("foo/bar")});
+  FSTAssertQueryDocumentMapping(4, expected_keys);
+}
+
+TEST_P(LocalStoreTest,
+       HandlesPatchMutationWithTransformationThenBundledDocuments) {
+  // Note: see comments in HandlesPatchMutationWithTransformThenRemoteEvent.
+  // The behavior for this and remote event is the same.
+  core::Query query = Query("foo");
+  AllocateQuery(query);
+
+  WriteMutation(testutil::PatchMutation(
+      "foo/bar", Map(), {testutil::Increment("sum", Value(1))}));
+
+  FSTAssertNotContains("foo/bar");
+  FSTAssertChanged(DeletedDoc("foo/bar"));
+
+  ApplyBundledDocuments({Doc("foo/bar", 1, Map("sum", 1337))});
+  FSTAssertChanged(
+      Doc("foo/bar", 1, Map("sum", 1), DocumentState::kLocalMutations));
+  FSTAssertContains(
+      Doc("foo/bar", 1, Map("sum", 1), DocumentState::kLocalMutations));
+
+  DocumentKeySet expected_keys({Key("foo/bar")});
+  FSTAssertQueryDocumentMapping(4, expected_keys);
+}
+
+TEST_P(LocalStoreTest, HandlesSavingAndCheckingBundleMetadata) {
+  BundleMetadata metadata("bundle", 1, SnapshotVersion(Timestamp(3, 0)));
+  EXPECT_FALSE(local_store_.HasNewerBundle(metadata));
+
+  local_store_.SaveBundle(metadata);
+
+  EXPECT_TRUE(local_store_.HasNewerBundle(metadata));
+}
+
+TEST_P(LocalStoreTest, HandlesSavingAndLoadingNamedQueries) {
+  core::Target target = Query("foo").ToTarget();
+
+  NamedQuery named_query(
+      "testQuery",
+      bundle::BundledQuery(std::move(target), core::LimitType::First),
+      SnapshotVersion(Timestamp::Now()));
+  local_store_.SaveNamedQuery(named_query, DocumentKeySet());
+
+  EXPECT_EQ(local_store_.GetNamedQuery("testQuery"), named_query);
+}
+
+TEST_P(LocalStoreTest,
+       SavingNamedQueriesAllocatesTargetsAndUpdatesTargetDocumentMapping) {
+  ApplyBundledDocuments({Doc("foo1/bar", 1, Map("sum", 1337)),
+                         Doc("foo2/bar", 1, Map("sum", 42))});
+  FSTAssertChanged(Doc("foo1/bar", 1, Map("sum", 1337)),
+                   Doc("foo2/bar", 1, Map("sum", 42)));
+  FSTAssertContains(Doc("foo1/bar", 1, Map("sum", 1337)));
+  FSTAssertContains(Doc("foo2/bar", 1, Map("sum", 42)));
+
+  core::Target target_1 = Query("foo1").ToTarget();
+
+  NamedQuery named_query_1(
```

Comment:
Nit: other local variables in this file that have numbers in their name don't use an additional underscore. I.e., this should be `named_query1`. Applies below as well.

Reply:
Fixed.